cli/sidecar/sidecar.go

```diff
@@ -155,16 +155,17 @@ func ConfigureSidecar(args []string) []string {
 		sidecarArgs = append(sidecarArgs, fmt.Sprintf("--cache_dir=%s", diskCacheDir))
 	}
 
+	// If there aren't any sidecar arguments, we don't need to spin up a sidecar at all.
+	if len(sidecarArgs) == 0 {
+		return args
+	}
+
 	if synchronousWriteFlag == "1" || synchronousWriteFlag == "true" {
 		sidecarArgs = append(sidecarArgs, "--synchronous_write")
 		sidecarArgs = append(sidecarArgs, "--bes_synchronous")
 		args = append(args, "--bes_upload_mode=wait_for_upload_complete")
```

Comment:
Unrelated to this PR, but it'd be nice if we could enable this flag whenever CI=true

Reply:
Not a bad idea, let me think about the best way to do this. 

I think a suggestion in the UI could be a good idea because then it helps non-CLI folks as well